packages/concerto-core/lib/decoratormanager.js

```diff
@@ -252,6 +252,186 @@ class DecoratorManager {
         newModelManager.fromAst(decoratedAst);
         return newModelManager;
     }
+    /**
+    * Adds a key-value pair to a dictionary (object) if the key exists,
+    * or creates a new key with the provided value.
+    *
+    * @param {Object} dictionary - The dictionary (object) to which to add the key-value pair.
+    * @param {string} key - The key to add or update.
+    * @param {any} value - The value to add or update.
+    * @param {string} declaration - The target decl.
+    * @param {string} property The target property.
+    */
```

Comment:
```suggestion
    * @private
    */
```

I don't believe that this should be publically addressable?